File: mq/src/share/cclient/io/PacketFlag.cpp
/*
 * Copyright (c) 2000, 2017 Oracle and/or its affiliates. All rights reserved.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0, which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * This Source Code may also be made available under the following Secondary
 * Licenses when the conditions for such availability set forth in the
 * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
 * version 2 with the GNU Classpath Exception, which is available at
 * https://www.gnu.org/software/classpath/license.html.
 *
 * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 */

/*
 * @(#)PacketFlag.cpp	1.4 06/26/07
 */ 

#include "PacketFlag.hpp"

/**
 *
 */
const char * 
PacketFlag::isQueueStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_IS_QUEUE) {
    return "IS_QUEUE ";
  } else {
    return "";
  }
}


/**
 *
 */
const char * 
PacketFlag::isRedeliveredStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_REDELIVERED) {
    return "REDLIVERED ";
  } else {
    return "";
  }
}



/**
 *
 */
const char * 
PacketFlag::isPersistentStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_PERSISTENT) {
    return "PERSISTENT ";
  } else {
    return "";
  }
}

/**
 *
 */
const char * 
PacketFlag::isSelectorsProcessedStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_SELECTORS_PROCESSED) {
    return "SELECTORS_PROCESSED ";
  } else {
    return "";
  }
}

/**
 *
 */
const char * 
PacketFlag::isSendAckStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_SEND_ACK) {
    return "SEND_ACK ";
  } else {
    return "";
  }
}

/**
 *
 */
const char * 
PacketFlag::isLastMessageStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_LAST_MESSAGE) {
    return "LAST_MESSAGE ";
  } else {
    return "";
  }
}

/**
 *
 */
const char * 
PacketFlag::isFlowPausedStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_FLOW_PAUSED) {
    return "FLOW_PAUSED ";
  } else {
    return "";
  }
}

/**
 *
 */
const char *
PacketFlag::isPartOfTransactionStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_PART_OF_TRANSACTION) {
    return "PART_OF_TRANSACTION ";
  } else {
    return "";
  }
}

/**
 *
 */
const char *
PacketFlag::isConsumerFlowPausedStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_CONSUMER_FLOW_PAUSED) {
    return "CONSUMER_FLOW_PAUSED ";
  } else {
    return "";
  }
}

/**
 *
 */
const char *
PacketFlag::isServerPacketStr(const PRUint16 bitFlags)
{
  if (bitFlags & PACKET_FLAG_SERVER_PACKET) {
    return "SERVER_PACKET ";
  } else {
    return "";
  }
}

File: mq/main/comm-io/src/main/java/com/sun/messaging/jmq/io/PacketFlag.java
/*
 * Copyright (c) 2000, 2017 Oracle and/or its affiliates. All rights reserved.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0, which is available at
 * http://www.eclipse.org/legal/epl-2.0.
 *
 * This Source Code may also be made available under the following Secondary
 * Licenses when the conditions for such availability set forth in the
 * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
 * version 2 with the GNU Classpath Exception, which is available at
 * https://www.gnu.org/software/classpath/license.html.
 *
 * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 */

/*
 * %W% %G%
 */ 

package com.sun.messaging.jmq.io;

/**
 * This class defines the JMQ packet bit flags and provides some
 * convenience routines.
 */
public class PacketFlag {

    public static final int Q_FLAG =  0x0001;
    public static final int R_FLAG =  0x0002;
    public static final int P_FLAG =  0x0004;
    public static final int S_FLAG =  0x0008;
    public static final int A_FLAG =  0x0010;
    public static final int L_FLAG =  0x0020;
    public static final int F_FLAG =  0x0040;
    public static final int T_FLAG =  0x0080;
    public static final int C_FLAG =  0x0100;
    public static final int B_FLAG =  0x0200;
    public static final int Z_FLAG =  0x0400;
    public static final int I_FLAG =  0x0800;
    public static final int W_FLAG =  0x1000;

    /**
     * Return a human readable string describing the bits set in "flags"
     */
    public static String getString(int flags) {
	String s =
	    ((flags & A_FLAG) == A_FLAG ? "A" : "") +
	    ((flags & S_FLAG) == S_FLAG ? "S" : "") +
	    ((flags & P_FLAG) == P_FLAG ? "P" : "") +
	    ((flags & R_FLAG) == R_FLAG ? "R" : "") +
	    ((flags & Q_FLAG) == Q_FLAG ? "Q" : "") +
	    ((flags & L_FLAG) == L_FLAG ? "L" : "") +
	    ((flags & F_FLAG) == F_FLAG ? "F" : "") +
	    ((flags & T_FLAG) == T_FLAG ? "T" : "") +
	    ((flags & B_FLAG) == B_FLAG ? "B" : "") +
	    ((flags & Z_FLAG) == Z_FLAG ? "Z" : "") +
	    ((flags & C_FLAG) == C_FLAG ? "C" : "") +
	    ((flags & I_FLAG) == I_FLAG ? "I" : "") +
	    ((flags & W_FLAG) == W_FLAG ? "W" : "");

        return s;
    }

}
